subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/http/HttpAuthenticationDependencyResolutionIntegrationTest.groovy

```diff
@@ -162,6 +164,48 @@ task listJars {
         'basic and digest' | 'authentication { basic(BasicAuthentication)\ndigest(DigestAuthentication) }' | DIGEST            | ['Basic', 'Digest']
     }
 
+    @Unroll
+    @Issue("gradle/gradle#5571")
+    public void "can resolve dependencies using #authSchemeName scheme from #authScheme authenticated with http header HTTP maven repository"() {
+        given:
+        def moduleA = mavenHttpRepo.module('group', 'projectA', '1.2').publish()
+        and:
+        buildFile << """
+repositories {
+    maven {
+        url "${mavenHttpRepo.uri}"
+        credentials(HttpHeaderCredentials) { header 'Name: Value' }
+        authentication { header(HttpHeaderAuthentication) }
+    }
+}
+configurations { compile }
+dependencies {
+    compile 'group:projectA:1.2'
+}
+task listJars {
+    doLast {
+        assert configurations.compile.collect { it.name } == ['projectA-1.2.jar']
+    }
+}
+"""
+
+        when:
+        serverAuthScheme = authScheme
+
+        and:
+        moduleA.pom.expectGet()
+        moduleA.artifact.expectGet()
+
+        then:
+        succeeds('listJars')
+        and:
+        server.authenticationAttempts.asList() == authenticationAttempts
```

Comment:
`server.authenticationAttempts` contains only `Authorization` header "prefixes", e.g. "Basic" for `Authorization: Basic ...`:

https://github.com/gradle/gradle/blob/1d58e09ddf1ccdf1a946e837878debcf8379533d/subprojects/internal-integ-testing/src/main/groovy/org/gradle/test/fixtures/server/http/HttpServerFixture.groovy#L158-L168

Reply:
I wouldn't hard-code the `TestHttpHeaderName` header in `HttpServerFixture`. Either make it configurable or, instead of using `authenticationAttempts`, record all headers of all requests and check the expectations in the tests.